scripts/upgrade

```diff
@@ -31,13 +30,30 @@ upgrade_type=$(ynh_check_app_version_changed)
 #=================================================
 ynh_script_progression --message="Ensuring downward compatibility..."
 
-# Fix is_public as a boolean value
-if [ "$is_public" = "Yes" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=1
-	is_public=1
-elif [ "$is_public" = "No" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=0
-	is_public=0
+# Cleaning legacy permissions
+is_public=$(ynh_app_setting_get --app=$app --key=is_public)
+skipped_uris=$(ynh_app_setting_get --app=$app --key=skipped_uris)
+unprotected_uris=$(ynh_app_setting_get --app=$app --key=unprotected_uris)
+protected_uris=$(ynh_app_setting_get --app=$app --key=protected_uris)
+
+# Remove is_public if exists
+if [ ! -z "$is_public" ]; then
+	ynh_app_setting_delete --app=$app --key=is_public
```

Comment:
Thanks, I recreated the upgrade script based on example_ynh's one (`permission` branch).
See the commit https://github.com/YunoHost-Apps/hextris_ynh/commit/845149815c9cbf004948ed51d9f5730ffef50c22 less than one minute ago ;-)

Reply:
```if [ ! -z $is_public ]; then
	if [ "$is_public" = "1" ]; then
		ynh_permission_update --permission "main" --add "visitors"
	fi
	ynh_app_setting_delete --app=$app --key=is_public
fi```
This script is overriding admin's settings when upgrading. If I install the app with is_public enabled and I finally decide to make it private (thru permissions management panel) the upgrade script shouldn't re-enable it.

Reply:
Yeah ! CI test passed !

Reply:
I remember that I had the same issue:
https://github.com/YunoHost/yunohost/blob/13b510b6249af25f2b0289df10a0b35baef90f7a/data/helpers.d/setting#L200

https://github.com/YunoHost-Apps/dotclear2_ynh/blob/110ef7e1d615e82928ed54ac7fe0aaf52b9f8dab/scripts/upgrade#L51-L76

In fact if you remove `skipped_uris`, the helper `ynh_app_setting_delete` will remove the `visitors` group from the permission. We did that to avoid another bug, there is no magical solution here sadly... The quick and dirty workaround I found it to re add the `visitors` group after deleting all keys from the legacy permission system.